metabase/metabase
# Add database form skeleton

Epic https://github.com/metabase/metabase/issues/26178
Demo https://www.loom.com/share/68d2d09ade364f79a4d0e8a944cb65e0

Adds the basic version of the database form built on `formik`:
- Reinitializes the form when the engine changes
- Allows field overrides with custom widgets
- Respects `required` field flag
- Respects `visible-if` rules

How to test:
- `yarn storybook` -> `DatabaseForm`
- Fill in required fields and submit the form

What doesn't work:
- Fields with `required` flag should be required only when `visible-if` matches.
- Metadata sync fields doesn't work. `SchedulePicker` crashes storybook.

<img width="848" alt="Screenshot 2022-11-28 at 15 54 45" src="https://user-images.githubusercontent.com/8542534/204294977-267ec34b-fc9e-406b-9bf8-6dfaede53313.png">
